src/client/components/AddReferenceLayerModal.tsx

```diff
@@ -0,0 +1,504 @@
+      !layerType && setFileError("Geojson must be point or polygon");
+      layerType &&
+        setCreateLayerResource({
+          data: { ...createLayerResource.data, layer_type: layerType }
+        });
+      layerType && setImportResource({ ...importResource, resource: importResponse });
+    }
+  }, [importResponse]);
+
+  const labelOptions = createLayerResource.data.fields
+    ? createLayerResource.data.fields.map(val => (
+        <option key={val} value={val}>
+          {val}
+        </option>
+      ))
+    : [];
+
+  return showModal ? (
+    <AriaModal
+      titleId="add-reference-layer-modal"
+      onExit={hideModal}
+      initialFocus="#primary-action"
+      getApplicationNode={() => document.getElementById("root") as Element}
+      underlayStyle={{ paddingTop: "4.5rem" }}
+    >
+      <Box sx={style.modal}>
+        <React.Fragment>
+          <Box sx={style.header}>
+            <Heading as="h1" sx={style.heading} id="add-reference-layer-modal-header">
+              Add reference layer
+            </Heading>
+          </Box>
+          <Flex
+            as="form"
+            sx={{ flexDirection: "column" }}
+            onSubmit={(e: React.FormEvent) => {
+              e.preventDefault();
+              const validatedForm = validate(formData, importResource);
+              if (validatedForm.valid === true) {
+                setCreateLayerResource({ data: validatedForm, isPending: true });
+                const referenceLayer = {
+                  name: validatedForm.name,
+                  label_field: validatedForm.label_field,
+                  project: validatedForm.project,
+                  layer: validatedForm.layer,
+                  layer_type: validatedForm.layer_type
+                };
+
+                createReferenceLayer(referenceLayer)
+                  .then((refLayer: ReferenceLayerWithGeojson) => {
+                    setCreateLayerResource({ resource: refLayer, data: validatedForm });
+                    store.dispatch(projectReferenceLayersFetch(project.id));
+                  })
+                  .catch(showActionFailedToast);
+              }
+            }}
+          >
+            {"isPending" in importResource && importResource.isPending ? (
+              <span>
+                <Spinner variant="spinner.small" /> Uploading
+              </span>
+            ) : "resource" in importResource || fileError ? (
+              <Box>
+                {fileError ? (
+                  <Box sx={style.uploadError}>
+                    <b>Error: {fileError}</b>
+                  </Box>
+                ) : "resource" in importResource ? (
+                  <Box>
+                    <Box sx={style.uploadSuccess}>
+                      <b>Upload success</b>
+                    </Box>
+                    <Flex>Number of features: {createLayerResource.data.numberOfFeatures}</Flex>
+                    <Flex>Layer type: {createLayerResource.data.layer_type}</Flex>
```

Comment:
Let's match what is in the wireframes here, as well as in the label names below:
![image](https://user-images.githubusercontent.com/4432106/130276632-3feca2c4-8da1-42ca-a72f-db438654cfec.png)

```suggestion
                    <Box>
                      Upload successful with {createLayerResource.data.numberOfFeatures}  {createLayerResource.data.layer_type.toLowerCase()}s. Give your layer a name and choose which property to use for labels.
                    </Box>
```